### PR TITLE
Fix link in SECURITY.md

### PR DESCRIPTION
**Pull request by lehors**
Signed-off-by: Arnaud J Le Hors <lehors@us.ibm.com>

#### Type of change

- Documentation update

#### Description

Fix link to the wiki page which has moved.

### CODE REVIEW

**Comment by denyeart** [comment]
Hit integration test flake https://jira.hyperledger.org/browse/FAB-18305

**Comment by denyeart** [comment]
@lehors Thanks! Did the repolint pass in your local env?
In the future, would there be a job that runs that automatically checks this in CI?

**Comment by lehors** [comment]
> @lehors Thanks! Did the repolint pass in your local env?

Yes, it did:

```
$ repolinter .
Target directory: /Users/lehors/Projects/github.com/hyperledger/fabric
Axiom language failed to run with error: Unexpected end of JSON input
Lint:
✔ apache-license-file: Contains Apache License.*Version 2.0 (LICENSE)
✔ code-of-conduct-file: Contains https://wiki.hyperledger.org/community/hyperledger-project-code-of-conduct (CODE_OF_CONDUCT.md)
✔ security-file-matches: Contains https://wiki.hyperledger.org/display/SEC/Defect.Response (SECURITY.md)
✔ readme-file-exists: Found file (README.md)
✔ readme-references-license: Contains license (README.md)
✔ maintainers-file-exists: Found file (MAINTAINERS.md)
✔ contributing-file-exists: Found file (CONTRIBUTING.md)
✔ changelog-file-exists: Found file (CHANGELOG.md)
✔ test-directory-exists: Found file (core/testutil)
✔ integrates-with-ci: Found file (ci/azure-pipelines.yml)
✔ notice-file-exists: Found file (NOTICE)
⚠ source-license-headers-exist:
	⚠ docs/build/html/_static/doctools.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/documentation_options.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/jquery-3.2.1.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/searchtools.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/language_data.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/jquery-3.5.1.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/underscore.js: The first 5 lines do not contain the pattern(s): Copyright
	⚠ docs/build/html/_static/underscore-1.3.1.js: The first 5 lines do not contain the pattern(s): Copyright
	⚠ docs/build/html/_static/jquery.js: The first 5 lines do not contain the pattern(s): Copyright
	⚠ docs/build/html/_static/websupport.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/js/html5shiv-printshiv.min.js: The first 5 lines do not contain the pattern(s): Copyright
	⚠ docs/build/html/_static/js/html5shiv.min.js: The first 5 lines do not contain the pattern(s): Copyright
	⚠ docs/build/html/_static/js/modernizr.min.js: The first 5 lines do not contain the pattern(s): Copyright, License
	⚠ docs/build/html/_static/js/theme.js: The first 5 lines do not contain the pattern(s): Copyright, License
✔ binaries-not-present: Excluded file type doesn't exist (**/*.exe,**/*.dll,!node_modules/**)
ℹ package-metadata-exists: ignored due to unsatisfied condition(s): "language=javascript"
ℹ package-metadata-exists: ignored due to unsatisfied condition(s): "language=ruby"
ℹ package-metadata-exists: ignored due to unsatisfied condition(s): "language=java"
✔ license-detectable-by-licensee: Licensee identified the license for project: Apache-2.0
ℹ package-metadata-exists: ignored due to unsatisfied condition(s): "language=python"

```

> In the future, would there be a job that runs that automatically checks this in CI?

I assume this will be done automatically indeed, I can't see how we would make sure this is run on a regular basis otherwise but I don't know whether we want to add that to CI, this may be something that's triggered from outside. @ryjones probably has some idea on that.

File: repolint.json
Comment (lindluni) [inline repolint.json]: Why do we include the unnecessary languages such as python, java, and javascript for this project? They simply report `ignored` so why include them at all given that we know what languages to expect.

File: repolint.json
Comment (lindluni) [inline repolint.json]: Same here

File: repolint.json
Comment (lindluni) [inline repolint.json]: The `source-license-headers-exist` check is looking for javascript files instead of `.go` files. This check also checks the `vendor` directory which needs to be excluded, haven't looked at the doc to see if there is a way to exclude this.

File: repolint.json
Comment (lindluni) [inline repolint.json]: We should change the `integrates-with-ci` check to only include `ci/azure-pipelines.yml` this is what we decided as our canonical location in Fabric for the file, so we would want the linter to alert us if a PR breaks from this convention.

**Review by lindluni — DISMISSED** [review]
A few comments and questions. Can we also format the JSON file properly, right now most of the lines are one lined which makes reading and updating difficult.

What would be the issue with simply adding it to CI on PR's? If this is the expected way a repo is meant to be setup, then I would expect that we should make sure that a PR doesn't deviate from that by running it at PR time.

Thanks

File: repolint.json
Comment (lindluni) [reply repolint.json]: You will also need to exclude directories containing `mock` and `fake` in them since these are generated, as well as `.pb.go` for generated protos

Comment (lindluni) [reply repolint.json]: 5 lines also isn't enough to capture the license in files that contain a go build constraint as the first line of the file

**Comment by denyeart** [comment]
lgtm complained with:
`[2021-02-01 14:50:00] [build-err] Only found JavaScript or TypeScript files that were empty or contained syntax errors.`

**Comment by lindluni** [comment]
> lgtm complained with:
> `[2021-02-01 14:50:00] [build-err] Only found JavaScript or TypeScript files that were empty or contained syntax errors.`

I need to get Ry to disable LGTM until we can get around to configuring it properly, to day we just take it at its default config, which does nothing for us save for finding some syntactical issues when they are fixed.

@ryjones Can you disable LGTM for the Fabric repo, only an Org Admin can do this, not a repo admin?

**Comment by denyeart** [comment]
> > lgtm complained with:
> > `[2021-02-01 14:50:00] [build-err] Only found JavaScript or TypeScript files that were empty or contained syntax errors.`
> 
> I need to get Ry to disable LGTM until we can get around to configuring it properly, to day we just take it at its default config, which does nothing for us save for finding some syntactical issues when they are fixed.
> 
> @ryjones Can you disable LGTM for the Fabric repo, only an Org Admin can do this, not a repo admin?

Yeah, I don't recall a discussion with the project maintainers to enable lgtm in the first place. If somebody has a compelling reason to enable them, we can have that discussion, but until there is some consensus on the topic, let's disable them.

**Comment by ryjones** [comment]
@denyeart this was enabled as advisory for all GitHub repos years ago. It never should have been a requirement, unless maintainers chose to enforce it. I've disabled it for now.

**Comment by lehors** [comment]
> A few comments and questions. Can we also format the JSON file properly, right now most of the lines are one lined which makes reading and updating difficult.
> 

Fair questions. I'll admit I merely copied the file from https://github.com/hyperledger-labs/hyperledger-community-management-tools/blob/main/repo_structure/repolint.json 

I think the original idea was to have the same config file used across Hyperledger but I have to say that I'm not convinced creating copies all over the place is a good idea. This is going to be a pain to maintain. At least, tailoring it for each repo would justify doing so rather than pointing to a global config file. From what I understand this used to be impossible but it is now.

> What would be the issue with simply adding it to CI on PR's? If this is the expected way a repo is meant to be setup, then I would expect that we should make sure that a PR doesn't deviate from that by running it at PR time.

I think it's merely a question of trade-off. But there is nothing prevent us from doing so if we don't mind paying the price of that extra check on every PR.

> Thanks

Thanks to you. Based on the various attempts by a few of us to implement the TSC decision so far I think this needs a bit more discussion. I'll put it on the TSC agenda.

**Comment by denyeart** [comment]
The saga continues in https://github.com/hyperledger/fabric/pull/2366

**Comment by lehors** [comment]
I reduced this PR to simply fixing the broken link in SECURITY.md so this can be merged without waiting for us to figure out the proper way of adding repolinter support.

**Comment by mastersingh24** [comment]
> Signed-off-by: Arnaud J Le Hors [lehors@us.ibm.com](mailto:lehors@us.ibm.com)
> 
> #### Type of change
> * New feature
> 
> #### Description
> Per the TSC, it was decided that all repositories must support repolinter and adhere to the common repository structure.
> 
> #### Additional details
> See https://wiki.hyperledger.org/display/TSC/Projects+have+two+quarters+to+comply+with+common+repo+structure

@lehors The description no longer matches the change

**Review by sykesm — APPROVED** [review]

**Comment by lehors** [comment]
> @lehors The description no longer matches the change

Fixed. Thanks.

**Pull request merged**
